drivers/sensors/apds9922.c

```diff
@@ -0,0 +1,2598 @@
+ *   Size of buffer read
+ *
+ ****************************************************************************/
+
+static ssize_t apds9922_als_read(FAR struct file *filep, FAR char *buffer,
+                                size_t buflen)
+{
+  FAR struct inode *inode;
+
+  FAR struct apds9922_dev_s *priv;
+  int *ptr;
+  int ret;
+
+  DEBUGASSERT(filep);
+
+  inode = filep->f_inode;
+  priv = inode->i_private;
+
+  DEBUGASSERT(inode && inode->i_private);
+
+  ret = nxmutex_lock(&priv->devlock);
+  if (ret < 0)
+    {
+      return ret;
+    }
+
+  if (buflen < 1)
+    {
+      snerr("ERROR: Buffer not large enough to read data\n");
+      return (ssize_t)-EINVAL;
+    }
+
+  ptr = (int *)buffer;
+
+  if (priv->als < 0)
+    {
+      *ptr = priv->als;
+    }
+  else
+    {
+      *ptr = apds9922_lux_calc(priv);
+    }
+
+  nxmutex_unlock(&priv->devlock);
+
+  return buflen;
+}
+
+/****************************************************************************
+ * Name: apds9922_als_write
+ *
+ * Description:
+ *   Standard character driver write method.
+ *
+ * Input Parameters:
+ *   filep - File structure pointer
+ *   buffer - Buffer to write
+ *   buflen - The write length of the buffer
+ *
+ * Returned Value:
+ *   -ENOSYS - this driver does not support the write method
+ *
+ ****************************************************************************/
+
+static ssize_t apds9922_als_write(FAR struct file *filep,
+                                  FAR const char *buffer, size_t buflen)
+{
+  return -ENOSYS;
+}
+
+/****************************************************************************
+ * Name: apds9922_als_ioctl
+ *
+ *  Description:
+ *    This routine is called when ioctl function call is performed for
+ *    the ambient light sensor of the apds9922 device.
+ *
+ * Input Parameters:
+ *   filep - file structure pointer.
+ *   cmd   - The IOCTL command.
+ *   arg - The argument of the IOCTL command.
+ *
+ * Returned Value:
+ *   Returns OK or a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+static int apds9922_als_ioctl(FAR struct file *filep, int cmd,
+                              unsigned long arg)
+{
+  FAR struct inode *inode = filep->f_inode;
+  FAR struct apds9922_dev_s *priv = inode->i_private;
+  int ret;
+  FAR uint8_t *ptr;
+
+  static struct apds9922_als_setup_s *als_setup;
+
+  ret = nxmutex_lock(&priv->devlock);
+  if (ret < 0)
+    {
+      return ret;
+    }
+
+  sninfo("cmd: 0x%" PRIx16 ", arg: %ld\n", cmd, arg);
+
+  switch (cmd)
+    {
+      case SNIOC_RESET:
+        ret = apds9922_reset(priv);
+        break;
+      case SNIOC_ALS_CONFIG:
+        als_setup = (struct apds9922_als_setup_s *)arg;
+        ret = apds9922_als_config(priv, als_setup);
+        break;
+      case SNIOC_GET_DEV_ID:
+        {
+          ptr = (FAR uint8_t *)arg;
+          DEBUGASSERT(ptr != NULL);
+          *ptr = priv->devid;
+          ret = OK;
+        }
```

Comment:
```suggestion
        ptr = (FAR uint8_t *)arg;
        DEBUGASSERT(ptr != NULL);
        *ptr = priv->devid;
        ret = OK;
```
`ret` can be only `OK` at this point, so the last assignment can be removed